src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-list/rgw-user-list.component.ts

```diff
@@ -134,6 +162,7 @@ export class RgwUserListComponent extends ListWithDetails {
     this.rgwUserService.list().subscribe(
       (resp: object[]) => {
         this.users = resp;
+        this.changeDetectorRef.detectChanges();
```

Comment:
Is this really needed?

Reply:
This was introduced similarly for bucket-list here https://github.com/ceph/ceph/pull/34606#discussion_r424444934. But as per our discussion instead tests should be adopted accordingly (using fakeAsync). So I will introduce the required test files changes for both buckets/users-list in this PR itself. And also will include minor improvement for stats parameter suggested here https://github.com/ceph/ceph/pull/36402#discussion_r591419092

Reply:
@avanthakkar We can get rid of that line and make the tests pass with a test refactoring like this:
```
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-list/rgw-user-list.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/rgw/rgw-user-list/rgw-user-list.component.spec.ts
@@ -1,6 +1,6 @@
 import { HttpClientTestingModule } from '@angular/common/http/testing';
 import { NO_ERRORS_SCHEMA } from '@angular/core';
-import { ComponentFixture, discardPeriodicTasks, fakeAsync, flush, TestBed, tick } from '@angular/core/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
 import { RouterTestingModule } from '@angular/router/testing';
 
@@ -8,7 +8,6 @@ import { of } from 'rxjs';
 
 import { RgwUserService } from '~/app/shared/api/rgw-user.service';
 import { TableActionsComponent } from '~/app/shared/datatable/table-actions/table-actions.component';
-import { CdTableFetchDataContext } from '~/app/shared/models/cd-table-fetch-data-context';
 import { SharedModule } from '~/app/shared/shared.module';
 import { configureTestBed, PermissionHelper } from '~/testing/unit-test-helper';
 import { RgwUserListComponent } from './rgw-user-list.component';
@@ -28,13 +27,16 @@ describe('RgwUserListComponent', () => {
   beforeEach(() => {
     rgwUserService = TestBed.inject(RgwUserService);
     rgwUserServiceListSpy = spyOn(rgwUserService, 'list');
-    rgwUserServiceListSpy.and.returnValue(of(null));
+    rgwUserServiceListSpy.and.returnValue(of([]));
     fixture = TestBed.createComponent(RgwUserListComponent);
     component = fixture.componentInstance;
+    spyOn(component, 'timeConditionReached').and.stub();
+    fixture.detectChanges();
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
+    expect(rgwUserServiceListSpy).toHaveBeenCalledTimes(1);
   });
 
   it('should test all TableActions combinations', () => {
@@ -79,7 +81,7 @@ describe('RgwUserListComponent', () => {
     });
   });
 
-  it('should test if rgw-user data is tranformed correctly', fakeAsync(() => {
+  it('should transform user data correctly', () => {
     rgwUserServiceListSpy.and.returnValue(
       of([
         {
@@ -96,9 +98,8 @@ describe('RgwUserListComponent', () => {
         }
       ])
     );
-    const context: CdTableFetchDataContext = null;
-    component.getUserList(context);
-    tick(10000);
+    component.getUserList(null);
+    expect(rgwUserServiceListSpy).toHaveBeenCalledTimes(2);
     expect(component.users).toEqual([
       {
         user_id: 'testid',
@@ -113,9 +114,9 @@ describe('RgwUserListComponent', () => {
         }
       }
     ]);
-  }));
+  });
 
-  it('should usage bars only if quota enabled', fakeAsync(() => {
+  it('should show usage bars only if quota enabled', () => {
     rgwUserServiceListSpy.and.returnValue(
       of([
         {
@@ -132,16 +133,14 @@ describe('RgwUserListComponent', () => {
         }
       ])
     );
-    const context: CdTableFetchDataContext = null;
-    component.getUserList(context);
-    tick(10000);
-    // flush();
-    // discardPeriodicTasks();
+    component.getUserList(null);
+    expect(rgwUserServiceListSpy).toHaveBeenCalledTimes(2);
+    fixture.detectChanges();
     const usageBars = fixture.debugElement.nativeElement.querySelectorAll('cd-usage-bar');
     expect(usageBars.length).toBe(2);
-  }));
+  });
 
-  it('should not show any usage bars if quota disabled', fakeAsync(() => {
+  it('should not show any usage bars if quota disabled', () => {
     rgwUserServiceListSpy.and.returnValue(
       of([
         {
@@ -158,11 +157,10 @@ describe('RgwUserListComponent', () => {
         }
       ])
     );
-    const context: CdTableFetchDataContext = null;
-    component.getUserList(context);
-    flush();
-    discardPeriodicTasks();
+    component.getUserList(null);
+    expect(rgwUserServiceListSpy).toHaveBeenCalledTimes(2);
+    fixture.detectChanges();
     const usageBars = fixture.debugElement.nativeElement.querySelectorAll('cd-usage-bar');
     expect(usageBars.length).toBe(0);
-  }));
+  });
 });
```

Reply:
Done! Thanks @alfonsomthd